In [1]:
# ==========================================
# PRAKTIKUM: BAYESIAN INFERENCE UNTUK QA TESTING
# ==========================================

print("🔍 Sistem Deteksi Akar Masalah (Bayesian QA)\n")

# 1. Deklarasi Data Historis
p_bug = 0.1  # P(Bug)
p_no_bug = 1 - p_bug  # P(Tidak Bug)

p_timeout_given_bug = 0.8  # P(Timeout | Bug)
p_timeout_given_no_bug = 0.05  # P(Timeout | Tidak Bug)

# 2. Menghitung P(Timeout) total (Hukum Probabilitas Total)
# Timeout bisa terjadi karena Bug ATAU karena bukan Bug (Jaringan jelek)
p_timeout = (p_timeout_given_bug * p_bug) + (p_timeout_given_no_bug * p_no_bug)

# 3. Menerapkan Teorema Bayes
# P(Bug | Timeout) = [ P(Timeout | Bug) * P(Bug) ] / P(Timeout)
p_bug_given_timeout = (p_timeout_given_bug * p_bug) / p_timeout

print("Laporan Insiden: Muncul laporan 'Timeout' dari Beta Tester!")
print("-" * 50)
print(f"Berdasarkan data historis Alpha/Beta Testing:")
print(f"Probabilitas bahwa Timeout ini disebabkan oleh BUG KODE adalah: {p_bug_given_timeout:.2%} \n")

# 4. Pengambil Keputusan (Decision Making)
if p_bug_given_timeout > 0.5:
    print("🚨 KEPUTUSAN AGEN AI: Tunda rilis aplikasi! Segera periksa kode sumber.")
else:
    print("✅ KEPUTUSAN AGEN AI: Abaikan sementara. Kemungkinan besar hanya masalah koneksi internet tester.")

🔍 Sistem Deteksi Akar Masalah (Bayesian QA)

Laporan Insiden: Muncul laporan 'Timeout' dari Beta Tester!
--------------------------------------------------
Berdasarkan data historis Alpha/Beta Testing:
Probabilitas bahwa Timeout ini disebabkan oleh BUG KODE adalah: 64.00% 

🚨 KEPUTUSAN AGEN AI: Tunda rilis aplikasi! Segera periksa kode sumber.


<p>Kesimpulan dari kodingan tersebut adalah kode tersebut merupakan sebuah contoh sederhana dalam mengimplementasikan teorema bayes untuk memutuskan apakah timeout terjadi karena bug kode atau tidak, juga untuk memutuskan dalam perilisan aplikasi. Dari hasilnya yaitu didapatkan bahwa timeout terjadi disebabkan oleh bug kode sebesar 64%, dan pada tahapan decision makingnya itu juga bug nya lebih dari 50%, maka perilisan aplikasi ditunda terlebih dahulu dan memperbaiki kembali kode nya </p>